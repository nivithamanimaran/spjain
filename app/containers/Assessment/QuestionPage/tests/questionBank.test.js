import React from 'react';
import {
  render,
  fireEvent,
  within,
  queryByAttribute,
  getByTestId,
  cleanup,
  screen,
} from 'react-testing-library';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../utils/themeProvider';
import QuestionBank from '../index';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Question bank', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <QuestionBank />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBank />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('question-bank')).toMatchSnapshot();
  });
});
