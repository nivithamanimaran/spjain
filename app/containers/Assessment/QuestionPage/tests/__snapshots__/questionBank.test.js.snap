// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question bank QuestionBank should renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-24"
  data-testid="question-bank"
>
  <div
    class="makeStyles-root-16"
  >
    <header
      class="MuiPaper-root MuiAppBar-root makeStyles-appBar-19 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-regular-21 MuiToolbar-gutters makeStyles-gutters-20"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-17 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <p
              class="MuiTypography-root makeStyles-titleIcon-3 MuiTypography-body1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-18 MuiTypography-h6"
        >
          <div
            class="MuiBox-root MuiBox-root-25 makeStyles-headWrap-1"
          >
            <p
              class="MuiTypography-root makeStyles-assessmentTitle-2 MuiTypography-body1"
            >
              Periodic assessment
            </p>
            <p
              class="MuiTypography-root makeStyles-titleIcon-3 MuiTypography-body1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41z"
                />
              </svg>
            </p>
            <a
              href="/assessment"
            >
              <p
                class="MuiTypography-root makeStyles-headerIcons-4 MuiTypography-body1"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                  />
                </svg>
              </p>
            </a>
            <p
              class="MuiTypography-root makeStyles-headerIcons-4 MuiTypography-body1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm0 8c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm-6 4c.22-.72 3.31-2 6-2 2.7 0 5.8 1.29 6 2H9zm-3-3v-3h3v-2H6V7H4v3H1v2h3v3z"
                />
              </svg>
            </p>
          </div>
        </h6>
        <div
          class="MuiBox-root MuiBox-root-26 makeStyles-headWrap-1"
        >
          <p
            class="MuiTypography-root makeStyles-discardBtnWrap-6 MuiTypography-body1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text KenButton__KenButtonDefault-sc-1sg8ful-0 dpKZcv MuiButton-textPrimary"
              data-testid="button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <p
                  class="MuiTypography-root makeStyles-discardLabel-10 MuiTypography-body1"
                  data-testid="discard-btn"
                >
                  Discard
                </p>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </p>
          <p
            class="MuiTypography-root makeStyles-btnWrap-5 MuiTypography-body1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined KenButton__KenButtonDefault-sc-1sg8ful-0 KenButton__KenButtonSecondary-sc-1sg8ful-1 juGKMU MuiButton-outlinedPrimary"
              data-testid="save-assessment-btn"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Save Assessment
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained KenButton__KenButtonDefault-sc-1sg8ful-0 dpKZcv MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled Mui-disabled"
              data-testid="button"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <p
                  class="MuiTypography-root makeStyles-upArrow-7 MuiTypography-body1"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-upArrowIcon-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                    />
                  </svg>
                </p>
                <p
                  class="MuiTypography-root makeStyles-saveButton-9 MuiTypography-body1"
                  data-testid="save-publish-btn"
                >
                  Save and Publish
                </p>
              </span>
            </button>
          </p>
        </div>
      </div>
    </header>
  </div>
  <div />
</div>
`;

exports[`Question bank should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-24"
  data-testid="question-bank"
>
  <div
    class="makeStyles-root-16"
  >
    <header
      class="MuiPaper-root MuiAppBar-root makeStyles-appBar-19 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-regular-21 MuiToolbar-gutters makeStyles-gutters-20"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-17 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <p
              class="MuiTypography-root makeStyles-titleIcon-3 MuiTypography-body1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-18 MuiTypography-h6"
        >
          <div
            class="MuiBox-root MuiBox-root-25 makeStyles-headWrap-1"
          >
            <p
              class="MuiTypography-root makeStyles-assessmentTitle-2 MuiTypography-body1"
            >
              Periodic assessment
            </p>
            <p
              class="MuiTypography-root makeStyles-titleIcon-3 MuiTypography-body1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41z"
                />
              </svg>
            </p>
            <a
              href="/assessment"
            >
              <p
                class="MuiTypography-root makeStyles-headerIcons-4 MuiTypography-body1"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                  />
                </svg>
              </p>
            </a>
            <p
              class="MuiTypography-root makeStyles-headerIcons-4 MuiTypography-body1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm0 8c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm-6 4c.22-.72 3.31-2 6-2 2.7 0 5.8 1.29 6 2H9zm-3-3v-3h3v-2H6V7H4v3H1v2h3v3z"
                />
              </svg>
            </p>
          </div>
        </h6>
        <div
          class="MuiBox-root MuiBox-root-26 makeStyles-headWrap-1"
        >
          <p
            class="MuiTypography-root makeStyles-discardBtnWrap-6 MuiTypography-body1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text KenButton__KenButtonDefault-sc-1sg8ful-0 dpKZcv MuiButton-textPrimary"
              data-testid="button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <p
                  class="MuiTypography-root makeStyles-discardLabel-10 MuiTypography-body1"
                  data-testid="discard-btn"
                >
                  labels:Discard
                </p>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </p>
          <p
            class="MuiTypography-root makeStyles-btnWrap-5 MuiTypography-body1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined KenButton__KenButtonDefault-sc-1sg8ful-0 KenButton__KenButtonSecondary-sc-1sg8ful-1 juGKMU MuiButton-outlinedPrimary"
              data-testid="save-assessment-btn"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                labels:Save_Assessment
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained KenButton__KenButtonDefault-sc-1sg8ful-0 dpKZcv MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled Mui-disabled"
              data-testid="button"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <p
                  class="MuiTypography-root makeStyles-upArrow-7 MuiTypography-body1"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-upArrowIcon-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                    />
                  </svg>
                </p>
                <p
                  class="MuiTypography-root makeStyles-saveButton-9 MuiTypography-body1"
                  data-testid="save-publish-btn"
                >
                  labels:Save_and_Publish
                </p>
              </span>
            </button>
          </p>
        </div>
      </div>
    </header>
  </div>
  <div />
</div>
`;
